### PR TITLE
fix(vue-app): fix regression with `scrollToTop`

### PR DESCRIPTION
**Pull request by pi0**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

`vue-router@3.4.0` landed with vuejs/vue-router#3199 which is breaking `scrollToTop: false` as we need to call `triggerScroll` event on first page too otherwise it will be delayed until next navigation which causes jumping to top (or saved position of previous page)

PR also introducing some code refactors

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/vue-app/template/router.scrollBehavior.js [hunk 1]
```diff
@@ -47,8 +45,12 @@ export default function (to, from, savedPosition) {
 
   const nuxt = window.<%= globals.nuxt %>
 
-  // triggerScroll is only fired when a new component is loaded
-  if (to.path === from.path && to.hash !== from.hash) {
+  if (
+    // Page hash changes
+    (to.path === from.path && to.hash !== from.hash) ||
+    // Initial load (vuejs/vue-router#3199)
+    to === from
```
Comment (pi0) [inline packages/vue-app/template/router.scrollBehavior.js:52]: @posva Do you think it is a safe check for detecting initial load?

**Comment by pi0** [comment]
merging to fix ci and test on edge

File: test/dev/unicode-base.size-limit.test.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ describe('nuxt minimal vue-app bundle size limit', () => {
   it('should stay within the size limit range', async () => {
     const filter = filename => filename === 'vue-app.nuxt.js'
     const legacyResourcesSize = await getResourcesSize(distDir, 'client', { filter })
-    const LEGACY_JS_RESOURCES_KB_SIZE = 16.5
```
Comment (farnabaz) [inline test/dev/unicode-base.size-limit.test.js:23]: 🤣

File: packages/vue-app/template/router.scrollBehavior.js (hunk 1 above)
Comment (posva) [reply packages/vue-app/template/router.scrollBehavior.js:52]: not completely sure, we could also check if `from` is the `START` location that is exported in v4.

Comment (pi0) [reply packages/vue-app/template/router.scrollBehavior.js:52]: I tried exposing [START](https://github.com/CoolCyberBrain/vue-router/blob/v3.2.0/src/util/route.js#L52) ref but checking here, on initial load `from` and `to` are same references but not equal to START:

image

**Pull request merged**